### PR TITLE
Move Files in File Tree

### PR DESCRIPTION
**Pull request by SarcevicAntonio**

### CODE REVIEW

**Comment by paoloricciuti** [comment]
Will you use neodrag? I feel like drag and drop of filles should be handled in mobile too (open to discuss tho)

**Comment by SarcevicAntonio** [comment]
We can have something for mobile, but maybe not drag and drop? I feel it is almost always shitty. Don't know something else tho.

**Comment by SarcevicAntonio** [comment]
This is without neodrag. I'm just using browser drag 👀

**Comment by paoloricciuti** [comment]
> We can have something for mobile, but maybe not drag and drop? I feel it is almost always shitty. Don't know about something else tho.

I mean we could have a command but I feel like the UX would not be the best...but we can try

**Comment by paoloricciuti** [comment]
> i imagine using command would be cumbersome... but i mean its kind of a complicated thing to do without drag and drop... maybe drag and drop works fine mobile, so lets first try that and then see how it goes.

I don't think browser drag and drop works at all in mobile.

**Comment by SarcevicAntonio** [comment]
Why wouldn't it? https://mdn.github.io/dom-examples/drag-and-drop/copy-move-DataTransfer.html works on my phone

**Comment by paoloricciuti** [comment]
> Why wouldn't it? https://mdn.github.io/dom-examples/drag-and-drop/copy-move-DataTransfer.html works on my phone

This is the first time that I see drag and drop working on mobile...lol

**Comment by paoloricciuti** [comment]
Looks good to me, but i found a small bug: if you move a folder the tab is not rewritten not closed so you could endup with an open, non existent file.

Fix this and merge imho

P.s. could the moving logic that replace the tab be applied to normal rename too? it feels far more snappy

**Comment by SarcevicAntonio** [comment]
i finally updated the logic. could you look at it and give it a whirl before merge please? @paoloricciuti

File: src/lib/components/Editor.svelte [hunk 1]
```diff
@@ -73,17 +73,22 @@
...
-		webcontainer.read_file(current_tab).then((file) => {
-			code = file;
-		});
+		webcontainer
+			.read_file(tab)
+			.then((file) => {
+				code = file;
+			})
+			.catch(() => {
+				$current_tab = '';
```
[the remark this answers is not in the document]
Comment (paoloricciuti) [reply src/lib/components/Editor.svelte:90]: Mmm i think the effect of this is that whenever there's an error reading the tab it closes every single tab (at least is what i'm seing in the preview)

Also i think there's still a bit of weirdness when moving folders. For example if i have a folder structure like this

```bash
lib/
└── test/
    ├── a.svelte
    └── testb/
        └── b.svelte
```

and i move testb into lib

```bash
lib/
├── test/
│   └── a.svelte
└── testb/
    └── b.svelte
```

it closes all the tabs

[the remark this answers is not in the document]
Comment (SarcevicAntonio) [reply src/lib/components/Editor.svelte]: I removed the problematic `.catch()` we didn't really need in the first place. You say it's okay to merge?

**Comment by SarcevicAntonio** [comment]
I removed the problematic catch, but moving the tree like you mentioned makes it wonky again. i feel like I need to fix my approach here, not sure how tho

**Comment by paoloricciuti** [comment]
> I removed the problematic catch, but moving the tree like you mentioned creates unexistend files again. i feel like I need to fix my approach here, not sure how tho

I'd say we can merge to at least have it here but i would open an issue for the case i displayed just to keep track of this, i can try work on that later

**Pull request merged**